### PR TITLE
Fixes #20363 - Move ruby-openscap to external scripts

### PR DESCRIPTION
**Pull request by xprazak2**
Two things on my mind: 
* I create tempfile names based on what the files are used for, but since they get immediately unlinked, maybe uuid as a name is enough

* We may consider using [CommandTask](https://github.com/theforeman/smart-proxy/blob/develop/lib/proxy/util.rb#L6) for command execution

### CODE REVIEW

File: lib/smart_proxy_openscap/shell_wrapper.rb [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+        file.rewind
+        file
+      end
+
+      def in_filename
+        unique_filename
+      end
+
+      def out_filename
+        unique_filename
```
Comment (ares) [inline lib/smart_proxy_openscap/shell_wrapper.rb]: would it make sense to memoize this value?

File: lib/smart_proxy_openscap/shell_wrapper.rb [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+      def prepare_in_file(in_file_content)
+        return unless in_file_content
+        file = Tempfile.new(in_filename, "/var/tmp")
+        file.write in_file_content
+        file.rewind
+        file
+      end
+
+      def in_filename
+        unique_filename
```
Comment (ares) [inline lib/smart_proxy_openscap/shell_wrapper.rb]: would it make sense to memoize this value?

**Comment by xprazak2** [comment]
I made a change to unique file names as suggested.

File: lib/smart_proxy_openscap/policy_parser.rb [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+      def in_filename
+        super
+      end
+
+      def out_filename
+        "#{in_filename}json-"
+      end
+
+      def failure_message
+        "Failure when running script which generates html reports"
```
Comment (ares) [inline lib/smart_proxy_openscap/policy_parser.rb]: s/generates html reports/renders policy guide/

**Comment by xprazak2** [comment]
Updated.

**Comment by ares** [comment]
Thanks @xprazak2, merging

**Comment by ares** [comment]
Probably worth of 0.6.6 release?

**Comment by xprazak2** [comment]
I agree.

**Pull request merged**